tlienart/Franklin.jl
# Taxonomy (second attempt)

Alright folks, this brings **tags** (back) in. When I have gained more confidence that this works as expected I'll merge and it will be minor release `0.8`. This is one of the main missing feature that I really wanted to add in Franklin.

I'd like some help to check it has the right behaviour. Basic stuff is unit-tested but the overall working of the tags is a bit subtle in terms of cleaning things up when tags are added/removed while the server is running. 

### To test this

Just add `@def tags = [...]` on meaningful pages. The first run will force-add a basic `_layout/tag.html` page which will be the mould for all generate tag pages. You should tweak this so that it respects the style of the rest of your pages. In the future it will be part of templates. 
The function `{{taglist}}` could be replaced by your own function `{{customTaglist}}` by adding a `hfun_customTaglist` in your `utils.jl` but let's maybe not worry about that right now.

The basic one `{{taglist}}` just adds the list of pages corresponding to  the tag by reverse chronological order. 

The very important thing to note is that `_layout/tag.html` **must not depend on local page variable**. It can only:

* use global page variables
* use page variable via `{{fill var path/to/page}}` 

The only  exception is `fd_tag` which contains the tag name. This you can use in the title for instance. For convenience let's reproduce the default HTML with some comments:

```html
<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Tag: {{fill fd_tag}}</title>
</head>
<body>
  <div class="{{div_content}} tagpage">
    {{taglist}}
  </div>
</body>
</html>
```

* You can see the `{{fill fd_tag}}` this will fill the tag name
* You can  see the `{{taglist}}` function which will generate something like

```html
<ul>
  <li><a href="/path/to/page_more_recent/">Title if it exists</a></li>
  <li><a href="/path/to/page_less_recent/">Title if it exists</a></li>
</ul>
```

Finally you can still use `{{ispage /tag/tagname/}}...{{end}}` to change the style depending on which tag page you're on.
